Comments/Services/AnswerService.cs

```diff
@@ -61,31 +71,58 @@ public AnswerService(ConsultationsContext consultationsContext, IUserService use
-                return (rowsUpdated: 0, validate: new Validate(valid: false, unauthorised: true, message: $"Not logged in deleting answer id:{answerId}"));
+            if (!_currentUser.IsAuthenticated)
+                return (rowsUpdated: 0, validate: new Validate(valid: false, unauthenticated: true, message: $"Not logged in deleting answer id:{answerId}"));
 
             var answerInDatabase = _context.GetAnswer(answerId);
 
             if (answerInDatabase == null)
                 return (rowsUpdated: 0, validate: new Validate(valid: false, notFound: true, message: $"Answer id:{answerId} not found trying to delete answer for user id: {_currentUser.UserId} display name: {_currentUser.DisplayName}"));
 
-	        if (!answerInDatabase.CreatedByUserId.Equals(_currentUser.UserId))
-		        return (rowsUpdated: 0, validate: new Validate(valid: false, unauthorised: true, message: $"User id: {_currentUser.UserId} display name: {_currentUser.DisplayName} tried to delete answer id: {answerId}, but it's not their answer"));
-
-			_context.Answer.Remove(answerInDatabase);
+			var consultationId = ConsultationsUri.ParseConsultationsUri(answerInDatabase.Question.Location.SourceURI).ConsultationId;
+            if (_currentUser.IsAuthenticatedByOrganisationCookieForThisConsultation(consultationId))
+            {
+				if (!answerInDatabase.OrganisationUserId.HasValue || !_currentUser.IsAuthorisedByOrganisationUserId(answerInDatabase.OrganisationUserId.Value))
+					return (rowsUpdated: 0, validate: new Validate(valid: false, unauthenticated: false, unauthorised: true, message: $"Organisation cookie user tried to delete answer id: {answerId}, but it's not their answer"));
+			}
+            else //accounts auth
+            {
+				if (!answerInDatabase.CreatedByUserId.Equals(_currentUser.UserId))
+					return (rowsUpdated: 0, validate: new Validate(valid: false, unauthenticated: true, message: $"User id: {_currentUser.UserId} display name: {_currentUser.DisplayName} tried to delete answer id: {answerId}, but it's not their answer"));
+			}
+			
+            _context.Answer.Remove(answerInDatabase);
 
 			return (rowsUpdated: _context.SaveChanges(), validate: null);
 		}
 
         public (ViewModels.Answer answer, Validate validate) CreateAnswer(ViewModels.Answer answer)
         {
-            if (!_currentUser.IsAuthorised)
-                return (answer: null, validate: new Validate(valid: false, unauthorised: true, message: "Not logged in creating answer"));
-
-	        var status = _context.GetStatus(StatusName.Draft);
-	        var question = _context.GetQuestion(answer.QuestionId);
-
-            var answerToSave = new Models.Answer(answer.QuestionId, _currentUser.UserId, answer.AnswerText, answer.AnswerBoolean, question, status.StatusId, null);
+            if (!_currentUser.IsAuthenticated)
+                return (answer: null, validate: new Validate(valid: false, unauthenticated: true, message: "Not logged in creating answer"));
```

Comment:
Is the answer message displayed to the user? Could do with rewriting the message

Reply:
this error message should never be hit by the front-end. if you're not logged in, it should show you the login banner component.
the only way this error would happen is if the front-end had a bug in it, or a user used postman/curl etc to hit the api, without an authentication cookie.